Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -650,7 +651,8 @@ bool TryInitialize() {
                     [desiredSuperview addSubview:xrView];
                 }
                 
-                ARFrame* currentFrame = [session currentFrame];
+                currentFrame = [session currentFrame];
+                [currentFrame retain];
```

Comment:
It doesn't matter too much, but if you wanted to make this more symmetrical, you could pull these two lines out and put them before the `dispatch_sync` call. Then we are acquiring/retaining the `currentFrame` and releasing it from the same thread. I tested this out a couple weeks ago and as far as I could tell it worked correctly.

Reply:
That is what I originally had, if you think this is cleaner that is good with me.  I had put them like this so that the lifetime of currentFrame would be tied to the lifetime of the XRFrame here, but I was waffling between the two places.

Reply:
Oh oops read this comment wrong the first time.  Yeah that makes sense to keep it all on the same thread, will do.